### PR TITLE
Add Thai Person provider and update formatter for Thai credit cards

### PR DESCRIPTION
**Pull request by rainmirror1**

### CODE REVIEW

File: src/Faker/Provider/th_TH/Person.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+namespace Faker\Provider\th_TH;
+
+class Person extends \Faker\Provider\Person
+{
+    /**
+     * @var array Thai male first name
+     * @link http://www.amarinbabyandkids.com/app/childname/
+     */
+    protected static $firstNameMale = array(
+        'กชฐณณ', 'กชนนท์', 'กชมนณัฐ', 'กนกภพ', 'กมลวร', 'กรณ์วรรธน์', 'กรภณ', 'กรภัทร', 'กรฤต', 'กรวรรธ', 'กช', 'กฤตพรต',
```
Comment (fzaninotto) [inline src/Faker/Provider/th_TH/Person.php]: There is not enough data in this class for a good randomization. can you find a datasource with a larger data set?

Comment (rainmirror1) [reply src/Faker/Provider/th_TH/Person.php]: Got your point. But I've got no idea how many items would be good, any suggestions?

Comment (fzaninotto) [reply src/Faker/Provider/th_TH/Person.php]: about 300 entries

**Comment by pimjansen** [comment]
Closing this PR since it got stale. Please feel free to reopen if needed

**Pull request closed without merging**